src/wasm/wasm-s-parser.cpp

```diff
@@ -1125,46 +1126,58 @@ Expression* SExpressionWasmBuilder::makeConst(Element& s, WasmType type) {
   return ret;
 }
 
-
-Expression* SExpressionWasmBuilder::makeLoad(Element& s, WasmType type, bool isAtomic) {
-  const char *extra = strchr(s[0]->c_str(), '.') + 5; // after "type.load"
-  if (isAtomic) extra += 7; // after "type.atomic.load"
-  auto* ret = allocator.alloc<Load>();
-  ret->isAtomic = isAtomic;
-  ret->type = type;
-  ret->bytes = getWasmTypeSize(type);
-  if (extra[0] == '8') {
-    ret->bytes = 1;
-    extra++;
-  } else if (extra[0] == '1') {
-    if (extra[1] != '6') throw ParseException("expected load16");
-    ret->bytes = 2;
-    extra += 2;
-  } else if (extra[0] == '3') {
-    if (extra[1] != '2') throw ParseException("expected load32");
-    ret->bytes = 4;
-    extra += 2;
+static uint8_t parseMemBytes(const char** in, uint8_t fallback) {
+  uint8_t ret;
+  const char* s = *in;
+  if (s[0] == '8') {
+    ret = 1;
+    (*in)++;
+  } else if (s[0] == '1') {
+    if (s[1] != '6') throw ParseException("expected 16 for memop size");
+    ret = 2;
+    *in += 2;
+  } else if (s[0] == '3') {
+    if (s[1] != '2') throw ParseException("expected 32 for memop size");;
+    ret = 4;
+    *in += 2;
+  } else {
+    ret = fallback;
   }
-  ret->signed_ = extra[0] && extra[1] == 's';
+  return ret;
+}
+
+static size_t parseMemAttributes(Element& s, Address* offset, Address* align, Address fallback) {
   size_t i = 1;
-  ret->offset = 0;
-  ret->align = ret->bytes;
+  *offset = 0;
+  *align = fallback;
   while (!s[i]->isList()) {
     const char *str = s[i]->c_str();
     const char *eq = strchr(str, '=');
-    if (!eq) throw ParseException("no = in load attribute");
+    if (!eq) throw ParseException("missing = in memory attribute");
     eq++;
     if (str[0] == 'a') {
-      uint64_t align = atoll(eq);
-      if (align > std::numeric_limits<uint32_t>::max()) throw ParseException("bad align");
-      ret->align = align;
+      uint64_t a = atoll(eq);
```

Comment:
`a` and `o` are kinda iffy names, but they're very local and are probably fine. But they have the same initialization code and so this might be more clear as:

```c++
uint64_t value = atoll(eq);
if (value > std::numeric_limits<uint32_t>::max()) throw ParseException("memory attribute value too large");
if (str[0] == 'a') {
  *align = value;
} else if (str[0] == 'o') {
  *offset = value;
} else throw ParseException("bad memory attribute");
```